torch/utils/data/datapipes/iter/grouping.py

```diff
@@ -65,7 +66,11 @@ def __init__(self,
         self.batch_size = batch_size
         self.drop_last = drop_last
         self.length = None
-        self.wrapper_class = DataChunk
+        source_depth = getattr(datapipe, '_dp_nesting_depth', 0)
+        if source_depth is None:
+            source_depth = 0
+        self._dp_nesting_depth = source_depth + 1
```

Comment:
We then increment `_dp_nesting_depth` from the normal `groupby`. Is this part really required?

Reply:
Moved all nesting to the next PR, as they still requires some cleanups.